runtime/interpreter/interpreter.go

```diff
@@ -3564,7 +3567,11 @@ func (interpreter *Interpreter) authAccountBorrowFunction(addressValue AddressVa
 			// which reads the stored value
 			// and performs a dynamic type check
 
-			if reference.ReferencedValue(interpreter) == nil {
```

Comment:
Do we need to do the same for the other places like `capabilityBorrowFunction()`, `exportValueWithInterpreter`, etc ?

Reply:
What do you mean?

Reply:
There are a few other places as well (like: `capabilityBorrowFunction`, `exportValueWithInterpreter`) where the value is dereferenced (`reference.ReferencedValue()` is used). I was wondering whether do we need to panic in those places as well or continue to silently fail.

Reply:
Ah, yea that is a question worth considering. For the scope of this PR I think just changing `load`, `copy` and `borrow` makes sense, but if we want to extend this behavior to anywhere we query storage then we would need to change the behavior there was well.

Reply:
The original issue has mentioned about handling the capability burrow as well.
> The load, copy, and borrow functions, as well as the capability borrow function, currently return nil both when the storage location is empty...

That's why I was wondering.
Is the plan to handle that separately? or am I missing something.

Reply:
ah I missed that the capability borrow function was also mentioned here. I will add that to this PR.